File: app/src/main/java/com/cray/software/passwords/utils/ThemedActivity.java
package com.cray.software.passwords.utils;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.cray.software.passwords.interfaces.Module;

public abstract class ThemedActivity extends AppCompatActivity {

    @Nullable
    private ThemeUtil themeUtil;
    @Nullable
    private Prefs mPrefs;

    @Nullable
    protected Prefs getPrefs() {
        return mPrefs;
    }

    @Nullable
    protected ThemeUtil getThemeUtil() {
        return themeUtil;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mPrefs = Prefs.getInstance(this);
        themeUtil = ThemeUtil.getInstance(this);
        if (themeUtil != null) setTheme(themeUtil.getStyle());
        if (Module.isLollipop()) {
            getWindow().setStatusBarColor(themeUtil.getColor(themeUtil.colorPrimaryDark()));
        }
    }
}

File: app/src/main/java/com/cray/software/passwords/utils/ThemedActivity.kt
package com.cray.software.passwords.utils

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import org.koin.android.ext.android.inject

abstract class ThemedActivity : AppCompatActivity() {

    protected val themeUtil: ThemeUtil by inject()
    protected val prefs: Prefs by inject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTheme(themeUtil.style)
    }
}
